GNUmakefile

```diff
@@ -9,7 +9,17 @@ GOTOOLS = \
 	github.com/golangci/golangci-lint/cmd/golangci-lint@v1.45.2 \
 	github.com/hashicorp/lint-consul-retry@master
 
-PROTOC_VERSION=3.15.8
+###
+# BUF_VERSION can be either a valid string for "go install <module>@<version>"
+# or the string @DEV to imply use what is currently installed locally.
+###
+BUF_VERSION='v1.4.0'
+
+###
+# PROTOC_GEN_GO_GRPC_VERSION can be either a valid string for "go install <module>@<version>"
+# or the string @DEV to imply use what is currently installed locally.
+###
+PROTOC_GEN_GO_GRPC_VERSION="v1.2.0"
```

Comment:
Ah so this doesn't magically track with some dependency in `go.mod` anymore like the former one I guess.

Reply:
Unfortunately the github.com/grpc/grpc-go library which is the same thing as google.golang.org/grpc tags the protoc plugin and the library differently. As of now the latest library version is 1.36 but the latest plugin version is 1.2.0. There are compile time type assertions to ensure you are using a compatible version of the library though:

https://github.com/hashicorp/consul/blob/addd859bb3ceca45df8bd982ac0be008e9f92b5d/proto-public/pbserverdiscovery/serverdiscovery_grpc.pb.go#L16-L19